wagtail/models/reference_index.py

```diff
@@ -510,6 +513,16 @@ def describe_source_field(self):
         if isinstance(field, models.ManyToOneRel):
             child_field = field.related_model._meta.get_field(model_path_components[2])
             return capfirst(child_field.verbose_name)
+        elif isinstance(field, StreamField):
+            label = f"{capfirst(field.verbose_name)}"
+            block = field.stream_block
+            block_idx = 1
+            while isinstance(block, StreamBlock):
```

Comment:
As a note to pre-empt feature creep more than anything...

If we're intending to support recursing into blocks to any depth, then we probably ought to recognise StructBlock / ListBlock, and at that point the concept of 'following path components' should be made into a method on the Block API rather than matching individual block types here.

However, given that:

1) it's probably uncommon for people to nest StreamBlocks directly in StreamBlocks
2) if we explicitly write this code to _not_ go more than one level, we'd probably want an `if isinstance(block, StreamBlock)` here anyhow to cover the possibility that we might support other top-level blocks besides StreamBlock in the near future (see #9738)

...this code in its current form will in practice only descend one level most of the time, and changing it to actually enforce only going one level deep would remove a maybe-occasionally-useful feature for no real benefit. So for that reason, I'm happy to stick with this :-)